mongodb-advanced/components/people-service.js

```diff
@@ -0,0 +1,38 @@
+/*!
+ * Copyright (c) 2016-2017 Digital Bazaar, Inc. All rights reserved.
+ */
+export default function factory($http, brAlertService) {
```

Comment:
See this as an example of a simple service and how promises are implemented.  Note that $http returns a $q style promise which is distinct from an ES6 Promise: https://github.com/digitalbazaar/bedrock-angular-authn-password/blob/material/password-service.js

Reply:
I strongly suspect that I still do not have this correct but I hope closer. Of the three functions (get, save and delete only get really needs to pass data back to the component. I guess though that for save it would be nice to get the error in the component so that it does not close the modal.